addons/account_edi/models/account_edi_format.py

```diff
@@ -20,50 +20,136 @@ class AccountEdiFormat(models.Model):
     name = fields.Char()
     code = fields.Char()
     hide_on_journal = fields.Selection([('import_export', 'Import/Export'), ('import', 'Import Only')], default='import_export', help='used to hide this EDI format on journals')
+    has_web_service = fields.Boolean()
 
     _sql_constraints = [
         ('unique_code', 'unique (code)', 'This code already exists')
     ]
 
+    ####################################################
+    # Low-level methods
+    ####################################################
+
+    @api.model
+    def create(self, vals):
+        res = super().create(vals)
+
+        journals = self.env['account.journal'].search([('type', 'in', ('sale', 'purchase'))])
+        for journal in journals:
+            if res._enable_edi_on_journal_by_default(journal):
+                journal.edi_format_ids += res
```

Comment:
that's a winding way to do

```python
    def create(self, vals):
        res = super().create(vals)

        journals = self.env['account.journal'].search([('type', 'in', ('sale', 'purchase'))])
        for journal in res._get_journals_per_default():
                journal.edi_format_ids += res
```

Reply:
I have to keep `_enable_edi_on_journal_by_default` because it is called when a journal is created. In this case (the format is created), we want to check all the journals. I could create a method _get_journals_per_default which would just split the current `create` in two :
```
    @api.model
    def create(self, vals):
        res = super().create(vals)

        for journal in res.get_journals_per_default():
            if res._enable_edi_on_journal_by_default(journal):
                journal.edi_format_ids += res

        return res

    def get_journals_per_default(self):
        self.ensure_one()
        return self.env['account.journal'].search([('type', 'in', ('sale', 'purchase'))]).filtered(lambda j: self._enable_edi_on_journal_by_default(j))
```